tools/seal_oem_partition.go

```diff
@@ -0,0 +1,204 @@
+	const devName = "oemroot"
+	const veritysetupImgPath = "./veritysetup.img"
+	imageID, err := loadVeritysetupImage(veritysetupImgPath)
+	if err != nil {
+		return fmt.Errorf("cannot load veritysetup image at %q, error msg:(%v)", veritysetupImgPath, err)
+	}
+	log.Println("docker image for veritysetup loaded.")
+	if err := unmountOEMPartition(); err != nil {
+		return fmt.Errorf("cannot umount OEM partition, error msg:(%v)", err)
+	}
+	log.Println("OEM parititon unmounted.")
+	hash, salt, err := veritysetup(imageID, oemFSSize4K)
+	if err != nil {
+		return fmt.Errorf("cannot run veritysetup, input:oemFSSize4K=%d, "+
+			"error msg:(%v)", oemFSSize4K, err)
+	}
+	grubPath, err := mountEFIPartition()
+	log.Println("EFI parititon mounted.")
+	if err != nil {
+		return fmt.Errorf("cannot mount EFI partition (/dev/sda12), error msg:(%v)", err)
+	}
+	partUUID, err := partutil.GetPartUUID("/dev/sda8")
+	if err != nil {
+		return fmt.Errorf("cannot read partUUID of /dev/sda8")
+	}
+	if err := appendDMEntryToGRUB(grubPath, devName, partUUID, hash, salt, oemFSSize4K); err != nil {
+		return fmt.Errorf("error in appending entry to grub.cfg, input:oemFSSize4K=%d, "+
+			"error msg:(%v)", oemFSSize4K, err)
+	}
+	log.Println("kernel command line modified.")
+	if err := removeVeritysetupImage(imageID); err != nil {
+		return fmt.Errorf("cannot remove veritysetup container or image, error msg:(%v)", err)
+	}
+	log.Println("docker image and container for veritysetup removed.")
+	return nil
+}
+
+// loadVeritysetupImage loads the docker image of veritysetup.
+// return the image ID.
+func loadVeritysetupImage(imgPath string) (string, error) {
+	cmd := exec.Command("sudo", "docker", "load", "-i", imgPath)
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("error in loading docker image, "+
+			"input: imgPath=%q, error msg: (%v)", imgPath, err)
+	}
+	var idBuf bytes.Buffer
+	cmd = exec.Command("sudo", "docker", "images", "veritysetup:veritysetup", "-q")
+	cmd.Stdout = &idBuf
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("error in reading image ID, "+
+			"cmd:%q, error msg: (%v)", "sudo docker images veritysetup:veritysetup -q", err)
+	}
+	if idBuf.Len() == 0 {
+		return "", fmt.Errorf("image ID not found, "+
+			"input: imgPath=%q", imgPath)
+	}
+	imageID := idBuf.String()
+	return imageID[:len(imageID)-1], nil
+}
+
+// removeVeritysetupImage removes the container and docker image of veritysetup
+func removeVeritysetupImage(imageID string) error {
+	cmd := exec.Command("sudo", "docker", "rmi", imageID)
+	cmd.Stdout = os.Stdout
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("error in removing docker image, "+
+			"id=%q, error msg: (%v)", imageID, err)
+	}
+	return nil
+}
+
+// mountEFIPartition mounts the EFI partition (/dev/sda12)
+// and returns the path where grub.cfg is at.
+func mountEFIPartition() (string, error) {
+	var tmpDirBuf bytes.Buffer
+	cmd := exec.Command("mktemp", "-d")
```

Comment:
How about creating a directory at /mnt/stateful_partition/ and removing that in the cleanup?